mempool/src/pool/mod.rs

```diff
@@ -326,6 +335,7 @@ struct TxFinalizer<'a> {
     cur_time: Time,
     events_broadcast: &'a mut EventsBroadcast,
     work_queue: &'a mut WorkQueue,
+    called_during_reorg: bool,
```

Comment:
seems a bit hacky. Bool is hard to read from the outside and easy to miss if modifications required, also information that finalizer is used in a specific situation (reorg) is external. Would passing `Option<&mut EventsBroadcast>` or even `enum EventsMode {Silent, Broadcast(&mut EventsBroadcast)}` be more robust approach here?